### DISCUSSION TITLE
Can't seem to expand relation

### DISCUSSION THREAD

**Discussion by JosephTTD** [pocketbase/pocketbase#1118] [Q&A] [answered]
I've fetched my collection and used the expand query, it does successfully return my data but without the relation being expanded. I have searched for a similar discussion but nothing works really.
 
Here's my code 
```
async function getPrompts() {
 const db = new PocketBase("http://127.0.0.1:8090");
 const result = await db.records.getList("prompts", 1, 30, {expand: 'user'})

 return results
}
```
In the dashboard the query shows `/api/collections/prompts/records?page=1&perPage=30&expand=user` with a 200 status

Here's the logged result

```
{
  page: 1,
  perPage: 1,
  totalItems: 1,
  totalPages: 1,
  items: [
    n {
      id: 'RECORD_ID',
      created: '2022-11-25 11:30:36.378',
      updated: '2022-11-25 13:05:09.547',
      '@collectionId': 'COLLECTION_ID',
      '@collectionName': 'prompts',
      argument: 'Hello world here's a test!',
      slug: 'hello-world-heres-a-test',
      solved: false,
      user: 'RELATION_ID',
      '@expand': {}
    }
  ]
}
```
Thanks in advance

**Comment by ganigeorgiev** [comment, marked as answer]
From the examples it seems that you are using PocketBase v0.7.x.

Is `user` a `relation` or `user` type field. If the latter then it cannot be expanded. You can change your field to a `relation` pointing to the `profiles` collection. Also make sure that the related collection View API rule is properly configured.

---

With that said, I would recommend to update to PocketBase v0.8.0. The `User` models and `profiles` collection are now merged to avoid the above confusion.

For a v0.7.x -> v0.8.0 migration guide you can check https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#migrate-from-v07x.

You'll have to update your JS SDK version also to `^0.8.0`. Detailed changelog what has changed there you could find in https://github.com/pocketbase/js-sdk/blob/master/CHANGELOG.md

**Reply by ganigeorgiev** [reply]
_Additionally, if you haven't already, you could find information about expand and how it works in https://pocketbase.io/docs/expanding-relations/ (the docs are valid for PocketBase v0.8+ only)_

**Reply by JosephTTD** [reply]
I'm using the relation type field and it's already pointing to the profiles collection as you can see in the image below. However I am using `pocketbase: "^0.7.4",`. So I'll update that and see if that makes a difference. 

<img width="646" alt="Screenshot 2022-11-25 at 14 16 16" src="https://user-images.githubusercontent.com/20605906/204003828-d4e71447-2302-4fdc-a0bf-57c5e3d22d48.png">

**Reply by ganigeorgiev** [reply]
@JosephTTD If it is a relation then make sure that you have "View API rules" access in the `profiles` collection (in the old admin, this could be accessed via "Users > Edit profiles collection (settings cogwheel) > API Rules").

_But in general Yes, it would be better to update to PocketBase v0.8.0 (both the executable and the JS SDK). Don't forget to apply the manual migrations steps from the above linked changelogs._

**Reply by JosephTTD** [reply]
Thank you! It's because of the access in Profile section as well as updating the sdk version. Appreciate your help

**Reply by jacksonsophat** [reply]
When I use expand, is it possible to only pull in a few items (username, avatar and name) from the user collection? 
Thank you!

**Reply by ganigeorgiev** [reply]
@jacksonsophat 
At the moment no (_it is in the [roadmap](https://github.com/orgs/pocketbase/projects/2?pane=issue&itemId=16153865)_). Depending on the use case and the type of relations, alternatively you can create a [View collection](https://pocketbase.io/docs/collections/#view-collection) with subset of the original `base` collection fields.
